py/server/deephaven/experimental/s3.py

```diff
@@ -118,12 +145,25 @@ def __init__(self,
                     (access_key_id is None and secret_access_key is not None)):
                 raise DHError("Either both access_key_id and secret_access_key must be provided or neither")
 
+            credentials_used = []
             if access_key_id is not None:
-                if anonymous_access:
-                    raise DHError("Only one set of credentials may be used, requested both key and anonymous")
                 builder.credentials(_JCredentials.basic(access_key_id, secret_access_key))
-            elif anonymous_access:
+                credentials_used.append("access_key_id")
+
+            if anonymous_access:
                 builder.credentials(_JCredentials.anonymous())
+                credentials_used.append("anonymous_access")
+
+            if profile_credentials:
+                builder.credentials(_JCredentials.profile())
+                credentials_used.append("profile_credentials")
+
+            if aws_default_credentials:
+                builder.credentials(_JCredentials.awsDefaultCredentials())
+                credentials_used.append("aws_default_credentials")
+
+            if len(credentials_used) > 1:
+                raise DHError(f"Only one set of credentials may be used, requested {', '.join(credentials_used)}")
```

Comment:
I don't think we'll get a chance to raise this prettier error; as soon as builder.credentials is called more than once, an error will be thrown.